src/test-run/commands/index.js

```diff
@@ -1,25 +1,54 @@
-export const TYPE = {
-    click:     'click',
-    assertion: 'assertion',
-    testDone:  'test-done'
-};
-
-export function createClickCommand () {
-    return {
-        type:      TYPE.click,
-        arguments: {}
-    };
+import TYPE from './type';
+import Assignable from '../../utils/assignable';
+import { ActionSelectorError } from '../../errors/test-run';
+import { ClickOptions } from './options';
+
+// Validators
+function selector (option, val) {
+    if (typeof val !== 'string')
+        throw new ActionSelectorError(val);
 }
 
-export function createAssertionCommand () {
-    return {
-        type:      TYPE.assertion,
-        arguments: {}
-    };
+function clickOptions (option, val) {
+    return new ClickOptions(val, true);
 }
 
-export function createDoneCommand () {
-    return {
-        type: TYPE.testDone
-    };
+// Commands
+export class Click extends Assignable {
+    constructor (obj) {
+        super(obj);
+
+        this.type      = TYPE.click;
+        this.arguments = {
+            selector: null,
+            options:  {}
+        };
+
+        this._assignFrom(obj, true);
+
+        this.arguments.selector = `(function () { return document.querySelector('${this.arguments.selector}') })()`;
+    }
+
+    _getAssignableProperties () {
+        return [
+            { name: 'arguments.selector', type: selector },
+            { name: 'arguments.options', type: clickOptions }
+        ];
+    }
 }
+
+export class TestDone {
+    constructor () {
+        this.type = TYPE.testDone;
+    }
+}
+
+// Factory
+export function createFromObj (obj) {
```

Comment:
Maybe `createCommandFromObject`? Then you will not need this awkward alias in compiler.